This is a pull request for issue https://github.com/acenturyandabit/fortknight/issues/56

VishalVinayRam opened the pull request:
1. Things done in this PR:-

- Add progress to all the achievements that are made.
- All the achievements are made into a JSON file, which makes adding more achievements easier and allows for easy modification without touching the code.
- Only the categorization part was left as more details regarding how they need to be categorized were not provided.

2. Major changes are done in index.html, CSS, and achievement.js.
3. Special things added were:-

- The progress and achievements are separated into tab-like sections to reduce the unwanted scrolling to check for achievements.

Commit:
Setting  the ui

VishalVinayRam committed:
setting the data from json

VishalVinayRam committed:
adding progress into of achivements

VishalVinayRam committed:
adding count and of achievement

VishalVinayRam committed:
removing achiements.html

PushpenderSaini0 commented on js/achievements_highscores.js:
```diff
@@ -1,133 +1,74 @@
+}
+
+
+function initializeAchievements() {
+    for (const category in achievementList) {
+        for (const achievement in achievementList[category].achievements) {
+            const achData = achievementList[category].achievements[achievement];
+            if (!highscoreDict.achievementProgress[achievement]) {
+                highscoreDict.achievementProgress[achievement] = achData;
+            }
+            highscoreDict.achievementProgress[achievement].achievementName = achData.achievementName;
+            highscoreDict.achievementProgress[achievement].achievementDescription = achData.achievementDescrption;
+            highscoreDict.achievementProgress[achievement].achievementGoal = achData.achievementGoal;
+        }
+    }
+}
```
This can be removed !

PushpenderSaini0 commented on package-lock.json:
Why did we create a lock file ?

PushpenderSaini0 commented:
This branch is out of sync from the master please sync the changes.

PushpenderSaini0 commented on index.html:
```diff
@@ -203,6 +209,66 @@ <h2>Debug Informations</h2>
+
+
+        }
+        document.addEventListener('DOMContentLoaded', () => {
```
Why are you adding to the event again, this is already being done above, i believe these can be combined them

VishalVinayRam replied:
I had kept it combined but this made it easy for debugging and further development.

PushpenderSaini0 replied:
Lets extract these as separate function and then we can call by adding a listener, also any reason why we have this code in HTML file ?

PushpenderSaini0 commented on index.html:
```diff
@@ -154,32 +154,38 @@ <h3>FAQ</h3>
-                        -->
+                    <div class="view-toggle">
+                        <button id="viewByProgress">View by Progress</button>
+                        {% comment %} <button id="viewByCategory">View by Category</button> {% endcomment %}
```
What is `{% comment %}` , `{% endcomment %}` ?

PushpenderSaini0 commented:
Since you have made UI changed please also add the new UI screenshots in the PR description, this is helpful for other to review them.

VishalVinayRam commented:
![Screenshot_2025-02-04_22-28-06](https://github.com/user-attachments/assets/f228d424-cca8-466c-ad11-f952dc1a45fe)
These are some of the screenshots for the UI changes .

VishalVinayRam committed:
Removing some part of the code on suggestiosn

VishalVinayRam committed:
Merge branch 'acenturyandabit:master' into master

VishalVinayRam committed:
removing commented code

PushpenderSaini0 commented on js/achievements_highscores.js:
```diff
@@ -1,133 +1,61 @@
 };
 
+// Call this function to load the achievements
+loadAchievementsFromJson('achievements.json');
```
`achievements.json` does not match with the file name you have added !

VishalVinayRam committed:
Fixig ths json and some css

VishalVinayRam commented:
The file named is fixed

acenturyandabit commented on index.html:
```diff
@@ -48,7 +48,7 @@ <h2>Chess Battle Royale</h2>
                     <div style="text-align: center; display:block; padding: 5px 10px; background: rgba(0,0,0,0.7); color:white"
                         class="ach_modal_inner">
-                        <h3>Achievement Unlocked!</h3>
+                        <link href="/architement.html">Achievement Unlocked!</link>
```
This seems to be a broken link after you removed the architement.html file earlier.

acenturyandabit commented on js/main.js:
```diff
@@ -481,3 +481,13 @@ window.addEventListener('resize', () => {
         drawPlayer();
     }
 });
+
```
This code is no longer necessary since we removed the achievements.html page,

acenturyandabit commented on js/dom_events.js:
```diff
@@ -88,13 +88,13 @@ document.querySelector('.gmode').addEventListener('input', (e) => {
 });
 document.querySelector('.sidebar').addEventListener('click', (e) => {
-    let path = e.path || e.composedPath();
+    let path = e.composedPath();
```
The code you put in the index.html could go here.

acenturyandabit commented on index.html:
```diff
@@ -203,6 +198,39 @@ <h2>Debug Informations</h2>
         <h1> Move Timer: </h1>
         <progress value="0" max="2.5" id="timer-bar"></progress>
     </div>
+    <script>{
```
Let's not put this script inline, since no other scripts are inline. Instead, please move this to dom_events.js

acenturyandabit commented on js/achievements_highscores.js:
```diff
@@ -330,7 +260,6 @@ localStorage.setItem('fortknightHS', JSON.stringify(highscoreDict));
 
-let highscoreDict;
 try {
 highscoreDict = JSON.parse(localStorage.getItem('fortknightHS'));
```
This code seems to be a duplicate of line 43. Did you move the code and then forget to delete the copy?

VishalVinayRam replied:
Sorry have removed that

VishalVinayRam committed:
Removing the extra parts

VishalVinayRam committed:
Removing the comments

VishalVinayRam commented:
Are all the changes ok and are we good to merge this.

PushpenderSaini0 commented:
> Are all the changes ok and are we good to merge this.

https://github.com/acenturyandabit/fortknight/pull/118#discussion_r1956684806

PushpenderSaini0 reviewed (changes requested):
This PR breaks the UI, i am not able to move the knight, tested on Firefox, Chrome

VishalVinayRam committed:
Completly working code

VishalVinayRam committed:
Merge branch 'master' into secound-branch

VishalVinayRam committed:
Merge pull request #1 from VishalVinayRam/secound-branch

Completly working code

VishalVinayRam commented:
@PushpenderSaini0  thank you pointing it out , I removed some code making the UI messed up. I have fixed that as I had remove the UI component which was responsible for making it work. Now I have fixed that and hopefully all the changes aswell. 
![Screenshot_2025-02-23_00-01-23](https://github.com/user-attachments/assets/47e429eb-9f5a-4fe7-a83c-d5ba2cc7df77)
![Screenshot_2025-02-23_00-00-47](https://github.com/user-attachments/assets/360bd387-2944-4e0c-967e-820e9f2beec2)

PushpenderSaini0 commented on index.html:
```diff
@@ -147,39 +147,52 @@ <h3>Scoring:</h3>
                         <p style="color: #8CA1C5">+1 point for every turn on the field</p>
                         <p style="color: #467AD5">+2 points for every king taken</p>
-                        <p style="color: #CF469F">+3 points for every knight, bishop, rook taken</p>
+                        <p style="color: #CF469F">+3 points for everhttps://github.com/VishalVinayRam/fortknight/pull/1/conflict?name=js%252Fachievements_highscores.js&ancestor_oid=ff72e44f072b7d8e92e5c202b8a0660974c23942&base_oid=fc110027706e7a6bcfe7286e52c338b0ba0f737c&head_oid=3044cf891313c474cbabe050b86c1183d95c0185y knight, bishop, rook taken</p>
```
Why this change ?

VishalVinayRam commented:
I had to retace back to some old commits to find the correct and make. Ba
gws over it.

On Sun, 23 Feb, 2025, 01:56 Pushpender Saini, ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In index.html
> <https://github.com/acenturyandabit/fortknight/pull/118#discussion_r1966595228>
> :
>
> > @@ -147,39 +147,52 @@ <h3>Scoring:</h3>
>                      <div class="rules">
>                          <p style="color: #8CA1C5">+1 point for every turn on the field</p>
>                          <p style="color: #467AD5">+2 points for every king taken</p>
> -                        <p style="color: #CF469F">+3 points for every knight, bishop, rook taken</p>
> +                        <p style="color: #CF469F">+3 points for everhttps://github.com/VishalVinayRam/fortknight/pull/1/conflict?name=js%252Fachievements_highscores.js&ancestor_oid=ff72e44f072b7d8e92e5c202b8a0660974c23942&base_oid=fc110027706e7a6bcfe7286e52c338b0ba0f737c&head_oid=3044cf891313c474cbabe050b86c1183d95c0185y knight, bishop, rook taken</p>
>
> Why this change ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/acenturyandabit/fortknight/pull/118#pullrequestreview-2635247937>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVOMOGGVTPNAPYPL53NBABD2RDMQ7AVCNFSM6AAAAABWMX2NTOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMMZVGI2DOOJTG4>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>

VishalVinayRam commented:
These are showing points and achievements

On Sun, 23 Feb, 2025, 10:02 VISHAL VINAYRAM, ***@***.***>
wrote:

> I had to retace back to some old commits to find the correct and make. Ba
> gws over it.
>
> On Sun, 23 Feb, 2025, 01:56 Pushpender Saini, ***@***.***>
> wrote:
>
>> ***@***.**** commented on this pull request.
>> ------------------------------
>>
>> In index.html
>> <https://github.com/acenturyandabit/fortknight/pull/118#discussion_r1966595228>
>> :
>>
>> > @@ -147,39 +147,52 @@ <h3>Scoring:</h3>
>>                      <div class="rules">
>>                          <p style="color: #8CA1C5">+1 point for every turn on the field</p>
>>                          <p style="color: #467AD5">+2 points for every king taken</p>
>> -                        <p style="color: #CF469F">+3 points for every knight, bishop, rook taken</p>
>> +                        <p style="color: #CF469F">+3 points for everhttps://github.com/VishalVinayRam/fortknight/pull/1/conflict?name=js%252Fachievements_highscores.js&ancestor_oid=ff72e44f072b7d8e92e5c202b8a0660974c23942&base_oid=fc110027706e7a6bcfe7286e52c338b0ba0f737c&head_oid=3044cf891313c474cbabe050b86c1183d95c0185y knight, bishop, rook taken</p>
>>
>> Why this change ?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/acenturyandabit/fortknight/pull/118#pullrequestreview-2635247937>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AVOMOGGVTPNAPYPL53NBABD2RDMQ7AVCNFSM6AAAAABWMX2NTOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMMZVGI2DOOJTG4>
>> .
>> You are receiving this because you were assigned.Message ID:
>> ***@***.***>
>>
>

VishalVinayRam commented:
Hey  where you able to test it now.


On Sun, Feb 23, 2025 at 10:02 AM VISHAL VINAYRAM <
***@***.***> wrote:

> These are showing points and achievements
>
> On Sun, 23 Feb, 2025, 10:02 VISHAL VINAYRAM, ***@***.***>
> wrote:
>
>> I had to retace back to some old commits to find the correct and make. Ba
>> gws over it.
>>
>> On Sun, 23 Feb, 2025, 01:56 Pushpender Saini, ***@***.***>
>> wrote:
>>
>>> ***@***.**** commented on this pull request.
>>> ------------------------------
>>>
>>> In index.html
>>> <https://github.com/acenturyandabit/fortknight/pull/118#discussion_r1966595228>
>>> :
>>>
>>> > @@ -147,39 +147,52 @@ <h3>Scoring:</h3>
>>>                      <div class="rules">
>>>                          <p style="color: #8CA1C5">+1 point for every turn on the field</p>
>>>                          <p style="color: #467AD5">+2 points for every king taken</p>
>>> -                        <p style="color: #CF469F">+3 points for every knight, bishop, rook taken</p>
>>> +                        <p style="color: #CF469F">+3 points for everhttps://github.com/VishalVinayRam/fortknight/pull/1/conflict?name=js%252Fachievements_highscores.js&ancestor_oid=ff72e44f072b7d8e92e5c202b8a0660974c23942&base_oid=fc110027706e7a6bcfe7286e52c338b0ba0f737c&head_oid=3044cf891313c474cbabe050b86c1183d95c0185y knight, bishop, rook taken</p>
>>>
>>> Why this change ?
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/acenturyandabit/fortknight/pull/118#pullrequestreview-2635247937>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AVOMOGGVTPNAPYPL53NBABD2RDMQ7AVCNFSM6AAAAABWMX2NTOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMMZVGI2DOOJTG4>
>>> .
>>> You are receiving this because you were assigned.Message ID:
>>> ***@***.***>
>>>
>>

psainics commented:
There are a few comments not addressed, please have a look !

 https://github.com/acenturyandabit/fortknight/pull/118#discussion_r1966595228
 https://github.com/acenturyandabit/fortknight/pull/118#discussion_r1956684806

VishalVinayRam committed:
This change was by mistake so removing that

VishalVinayRam committed:
Merge pull request #2 from VishalVinayRam/secound-branch

This change was by mistake so removing that
